src/python/pants/scm/git.py

```diff
@@ -526,20 +527,21 @@ def _read_object(self, relpath, max_symlinks):
         # A git symlink is stored as a blob containing the name of the target.
         # Read that blob.
         object_type, path_data = self._read_object_from_repo(sha=obj.sha)
-        assert object_type == 'blob'
+        assert object_type == b'blob'
 
-        if path_data[0] == '/':
+        if path_data[0] == b'/':
           # Is absolute, thus likely points outside the repo.
           raise self.ExternalSymlinkException(self.rev, relpath)
 
-        link_to = os.path.normpath(os.path.join(parent_path, path_data))
+        path_data_unicode = ''.join(v.decode('utf-8') for v in path_data)
+        link_to = os.path.normpath(os.path.join(parent_path, path_data_unicode))
         if link_to.startswith('../') or link_to[0] == '/':
           # Points outside the repo.
           raise self.ExternalSymlinkException(self.rev, relpath)
 
         # Restart our search at the top with the new path.
         # Git stores symlinks in terms of Unix paths, so split on '/' instead of os.path.sep
-        components = link_to.split(SLASH) + components
+        components = link_to.split(SLASH.decode('utf-8')) + components
```

Comment:
Should change the constant instead. Despite being ostensibly public, it's only used locally.